src/Combinatorics/Matroids/quantum_automorphism_groups.jl

```diff
@@ -0,0 +1,228 @@
+        new_relation = u[j, i]*u[k, i]
+        push!(relations, new_relation)
+      end
+    end
+  end
+
+  #row and column sum relations
+  for i in 1:n
+    new_relation_row = -1
+    new_relation_col = -1
+    for k in 1:n
+      new_relation_row += u[i,k]
+      new_relation_col += u[k,i]
+    end
+    push!(relations, new_relation_row)
+    push!(relations, new_relation_col)
+
+  end
+  return ideal(relations)
+end
+
+@doc raw"""
+    _quantum_automorphism_group_indices(M::Matroid, structure::Symbol=:bases)
+
+Get the indices of the relations that define the quantum automorphism group of a matroid for a given structure.
+
+# Examples
+```jldoctest
+M = uniform_matroid(3,4)
+bases(M)
+idx = Oscar._quantum_automorphism_group_indices(M,:bases)
+length(idx)
+
+# output
+
+1920
+"""
+function _quantum_automorphism_group_indices(M::Matroid, structure::Symbol=:bases)
+  @req structure in [:bases, :circuits, :flats] "structure must be one of :bases, :circuits, :flats"
+
+  n = length(M)
+  grdSet = matroid_groundset(M)
+
+  b    = [[] for _ in 1:n]
+  nb   = [[] for _ in 1:n]
+  rels = [[] for _ in 1:n]
+
+  sets  = eval(structure)(M)
+  sizes = unique(map(x -> length(x), sets))
+  tempGrdSet = reduce(vcat,[grdSet for i in 1:n])
+
+  for size in sizes 
+    size == 0 && continue
+    powerSet = unique(sort.(Oscar.subsets(tempGrdSet,size)))
+
+    setsOfSize = filter(x->length(x)==size,sets)
+    nonSets = setdiff(powerSet,setsOfSize) 
+
+    for set in setsOfSize
+      append!(b[size],collect(Oscar.permutations(set)))
+    end
+    for nonset in nonSets
+      append!(nb[size],collect(Oscar.permutations(nonset)))
+    end
+    for set in b[size]
+      for nonset in nb[size]
+        rel = []
+        for i in 1:size
+          push!(rel,(set[i],nonset[i]))
+        end
+        push!(rels[size],rel)
+        rel = []
+        for i in 1:size
+          push!(rel,(nonset[i],set[i]))
+        end
+        push!(rels[size],rel)
+      end
+    end
+  end
+
+  rels = unique.(rels)
+  return Vector{Vector{Tuple{Int,Int}}}(reduce(vcat,rels))
+end
+
+@doc raw"""
+    quantum_automorphism_group(M::Matroid, structure::Symbol=:bases)
+
+Get the Ideal that defines the quantum automorphism group of a matroid for a given structure.
+
+# Examples
+
+```jldoctest
+M = uniform_matroid(3,4);
+qAut = quantum_automorphism_group(M,:bases);
+length(gens(qAut))
+# output
+
+2040
+```
+"""
+function quantum_automorphism_group(
+  M::Matroid,
+  structure::Symbol=:bases)
+
+  @req matroid_groundset(M) isa Vector{Int} "Matroid must have integer groundset"
+
+  relation_indices = _quantum_automorphism_group_indices(M,structure)
+  n = length(M)
+
+  SN = quantum_symmetric_group(n)
+  A = base_ring(SN)
+  u = permutedims(reshape(gens(A),(n,n)),[2,1])
+
+  new_relations = []
+
+  for relation in relation_indices
+    temp = one(A)
+    for gen in relation
+      temp = temp * u[gen[1], gen[2]]
+    end
+    push!(new_relations,temp)
+  end
```

Comment:
```suggestion
  new_relations = elem_type(A)[prod(gen -> u[gen[1], gen[2]], relation; init=one(A))) for relation in relation_indices]
```
not pushing all the time but doing a list comprehension is more efficient, as constantly pushing means that julia needs to realloc space multiple times

Reply:
The drawback is IMHO readability. The pushing isn't too bad, Julia amortizes this (and one give use `sizehint!` to minimize extra allocations).

But definitely should use `prod` :-).

Reply:
I changed it to what lars suggested.